### PR TITLE
Add EC key validation to default provider

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Depends on #10631

File: providers/implementations/keymgmt/ec_kmgmt.c [hunk 1]
```diff
@@ -730,6 +731,35 @@ int ec_set_params(void *key, const OSSL_PARAM params[])
...
+        ok = ok && EC_GROUP_check(EC_KEY_get0_group(eck), ctx);
+
+    if ((selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY) != 0)
+        ok = ok && ec_key_public_check(eck, ctx);
+
+    if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0)
+        ok = ok && ec_key_private_check(eck);
+
+    if ((selection & OSSL_KEYMGMT_SELECT_KEYPAIR) == OSSL_KEYMGMT_SELECT_KEYPAIR)
+        ok = ok && ec_key_pairwise_check(eck, ctx);
```
Comment (levitte) [inline providers/implementations/keymgmt/ec_kmgmt.c]: When calling `ec_key_pairwise_check()`, is it *necessary* to call `ec_key_public_check()` and `ec_key_private_check()` first, or do you end up doing double work in that case?

Comment (slontis) [reply providers/implementations/keymgmt/ec_kmgmt.c]: may be a tiny bit of overlap - but they are completely different tests.

Comment (levitte) [reply providers/implementations/keymgmt/ec_kmgmt.c]: Ok then

**Comment by slontis** [comment]
Thanks. Merged to master.

**Pull request closed without merging**